src/injector.ts

```diff
@@ -0,0 +1,237 @@
+import Web3 from 'web3';
+import { outputFileSync } from 'fs-extra';
+import path from 'path';
+import { findAddresses } from './address-db';
+
+const multihashes : any = require('multihashes');
+
+const save = outputFileSync;
+
+import {
+  cborDecode,
+  getBytecode,
+  recompile,
+  RecompilationResult,
+} from './utils';
+
+declare interface StringMap {
+  [key: string]: string;
+}
+
+export interface InjectorConfig {
+  infuraPID? : string
+}
+
+export default class Injector {
+  private chains : any;
+  private infuraPID : string;
+
+  /**
+   * Constructor
+   * @param {InjectorConfig = {}} config
+   */
+  public constructor(config : InjectorConfig = {}){
+    this.chains = {};
+    this.infuraPID = config.infuraPID || "891fe57328084fcca24912b662ad101f";
+    this.initChains();
+  }
+
+  /**
+   * Instantiates a web3 provider for all public ethereum networks via Infura.
+   * If environment variable TESTING is set to true, localhost:8545 is also available.
+   */
+  private initChains(){
+    for (const chain of ['mainnet', 'ropsten', 'rinkeby', 'kovan', 'goerli']){
+      this.chains[chain] = {};
+      this.chains[chain].web3 = new Web3(`https://${chain}.infura.io/v3/${this.infuraPID}`);
+    }
+
+    // For unit testing with testrpc...
+    if (process.env.TESTING){
```

Comment:
This should be moved to a variable in the Injector config and set via process.env in the unit tests and server.